### ISSUE TITLE
Cannot Initialize Library

### ISSUE THREAD

**Issue by kellertobi91** [KeitIG/museeks#145]
System: Win 10 64bit
Version: 0.6.0 (Downloaded from Webpage, extracted to homefolder)

Error in Dev-Console:
An error occured while reading G:/Media - Music/787f0f70 (2)/a1780a61-52ca3de8.mp3 id3 tags: Error: expected frame header but was not found

I added my Folder, klicked on "Refresh Library" and the progress-bar ran through (as the folder contains only around 1.7k files) and stopped at around 99%

as I have never developed on windows (using this system for gaming only) I cannot get inside of the server console.

**Comment by kellertobi91** [comment]
When closed and reopened, the files were there and I can play music.

**Comment by KeitIG** [comment]
If it happens again, in Settings/Advanced, can you toggle "dev mode" and tell me if there are any errors ?

**Comment by kellertobi91** [comment]
I do not get any further errors than the one I provided in my initial post. I looked for the errors in: 'Settings => Advanced => toggle dev mode" and then in the browser console on the tab console. there was only this yellow line.

I could investigate in this after my examns in oktober.

**Comment by KeitIG** [comment]
The `Error: expected frame header but was not found` error is due to #103, but it shouldn't block the library refresh. Does it always happen at the end ? So basically the library refresh never ends ?

If yes, there may be a problem with https://github.com/KeitIG/museeks/blob/master/src/js/actions/LibraryActions.js#L155 not being called

**Comment by KeitIG** [comment]
(poke @YurySolovyov)

**Comment by YurySolovyov** [comment]
> If yes, there may be a problem with https://github.com/KeitIG/museeks/blob/master/src/js/actions/LibraryActions.js#L155 not being called

Yeah, that could be the case.
Can anyone upload file giving this error somewhere like dropbox or drive? /cc @kellertobi91 @KeitIG

**Comment by kellertobi91** [comment]
Sorry, i cannot provide this special file as this is a concert recording from a band I do the lighting for. This file comes from and multitrack mp3 recorder and is a stereo mixdown.

The ffmpeg info of this file is:

```
Stram 0
  Type: Audio
  Codec: MPEG Audio layer 1/2 (mpga)
  Channels: Stereo
  44100 Hz
  128 kB/s
```

The file has a title and an artist id3- tag.

**Comment by sebastiaanfranken** [comment]
I have the same issue. Scanning/refreshing hangs at 99%. If I restart the application only 29 tracks (of about 50.000) are in the library. 

I opened the dev mode and went into the Console, which was spammed with messages like

> An error occurred while reading /home/sfranken/Muziek/<file>.mp3 id3 tags: Error could not find metadata header.

There are a few tracks in my library which are buggy (bad/crummy/missing metadata). All other music player (GNOME Music, Rhythmbox, Clementine and VLC) ignore those tracks or show them as "unknown author/unkown title".

**Comment by KeitIG** [comment]
@sebastiaanfranken could you send me one of these files ? I have a library with multiple files like that, and they are treated well, with "Unknown artis" etc if needed.

**Comment by KeitIG** [comment]
self-poke: probably a problem with the progress counter.

**Comment by YurySolovyov** [comment]
I found corrupt file in my dir and it does not prevent library updates.
Also, warning is coming from [`here`](https://github.com/KeitIG/museeks/blob/master/src/js/utils/utils.js#L237), and it is not actually an error, because we are still constructing metadata (yeah, it is mostly a stub data, but you get the point) in this case.

This does not mean there is no error at all, we just need some more info.

@KeitIG maybe it is worth to add some indication of which file(s) is inspected ATM, like counter, or pass filenames too, just to let people know that there is actual process going on.

**Comment by KeitIG** [comment]
Would it be possible to add a timeout: like after 60 seconds of inactivity, it dispatches the end of library refresh ?

**Comment by YurySolovyov** [comment]
Maybe, I'd like to solve the root issue though

**Comment by sebastiaanfranken** [comment]
@KeitIG How do you want me to send you one of the files? Attach it here?

**Comment by KeitIG** [comment]
You been sens me an e-mail (see my profile) or if it supported by Github, here :)

**Comment by sebastiaanfranken** [comment]
I've sent you an email with a track attached. Hope it helps.

**Comment by KeitIG** [comment]
Hey @sebastiaanfranken !

I tried your file, but it is a Flac and Museeks does not support Flac yet. So it must be something else. Do you know exactly what kind of different files are in your library ? (aac, mp3, flac, wav...)

**Comment by sebastiaanfranken** [comment]
I mostly have flac files. Some files are wav, and that's it. No MP3, aac or any other codecs. The few files I could see in Museeks were flacs (and the wav files, since those are loose files and get indexed first I think) and they plan fine as well.

Weird.. Need any more info from me?

**Comment by YurySolovyov** [comment]
@sebastiaanfranken museeks-playable file(s) that errors would be helpful

**Comment by sebastiaanfranken** [comment]
The original error is gone. The only error's I'm getting now is "Uncaught TypeError: Cannot read property 'duration' of undefined" in bundle.js, line 12. I'll send a (original) wav file to @KeitIG for him to test, as per original request.

**Comment by KeitIG** [comment]
I think I see where the bug is. I will make a fix today if I can reproduce it.

**Comment by KeitIG** [comment]
Hey everyone, a fix has been found and will be released with 0.7.0 :)
